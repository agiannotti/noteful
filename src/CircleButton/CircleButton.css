
.NavCircleButton {
  background-color: #cacaca;
  border-radius: 30px;
  border: 0;
  color: black;
  display: block;
  font-size: 12px;
  font-weight: bold;
  height: 60px;
  padding: 8px;
  text-align: center;
  text-decoration: none;
  width: 60px;
}

a.NavCircleButton {
  padding: 16px 8px 8px;
}

.NavCircleButton:hover {
  cursor: pointer;
}
