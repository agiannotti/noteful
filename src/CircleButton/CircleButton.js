import React from "react";
import "./CircleButton.css";
import PropTypes from "prop-types";

export default function NavCircleButton(props) {
  const { tag, className, childrenm, ...otherProps } = props;

  return React.createElement(
    props.tag,
    {
      className: ["NavCircleButton", props.className].join(" "),
      ...otherProps,
    },
    props.children
  );
}

NavCircleButton.defaultProps = {
  tag: "a",
};

NavCircleButton.propTypes = {
  children: PropTypes.array,
  className: PropTypes.string,
  tag: PropTypes.func,
};
