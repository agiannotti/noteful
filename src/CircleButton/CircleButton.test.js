import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CircleButton from "./CircleButton";

describe(`CircleButton component`, () => {
  const props = {
    tag: "a",
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders a button.CircleButton by default", () => {
    const wrapper = shallow(<CircleButton />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the circle button from props", () => {
    const wrapper = shallow(<CircleButton {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
