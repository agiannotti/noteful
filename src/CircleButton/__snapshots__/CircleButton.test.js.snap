// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleButton component renders a button.CircleButton by default 1`] = `
<a
  className="NavCircleButton "
/>
`;

exports[`CircleButton component renders the circle button from props 1`] = `
<a
  className="NavCircleButton test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</a>
`;
